File: backend/app/Models/Amenity.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;

class Amenity extends Model
{
    protected $fillable = ['key', 'name', 'name_ne', 'category', 'icon'];

    public function listings(): BelongsToMany
    {
        return $this->belongsToMany(PropertyListing::class, 'listing_amenities');
    }
}

File: mobile/lib/features/listings/data/models/amenity.dart
/// Mirrors `AmenityResource`.
class Amenity {
  Amenity({required this.id, required this.key, required this.name, this.nameNe, this.category, this.icon});

  factory Amenity.fromJson(Map<String, dynamic> json) {
    return Amenity(
      id: json['id'] as int,
      key: json['key'] as String,
      name: json['name'] as String,
      nameNe: json['name_ne'] as String?,
      category: json['category'] as String?,
      icon: json['icon'] as String?,
    );
  }

  final int id;
  final String key;
  final String name;
  final String? nameNe;
  final String? category;
  final String? icon;
}
